trunk/source/Library-TraceCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/tracecheckerutils/assertorders/AssertOrderUtils.java

```diff
@@ -0,0 +1,120 @@
+ * terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * The ULTIMATE TraceCheckerUtils Library is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with the
+ * ULTIMATE TraceCheckerUtils Library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Additional permission under GNU GPL version 3 section 7: If you modify the ULTIMATE TraceCheckerUtils Library,
+ * or any covered work, by linking or combining it with Eclipse RCP (or a modified version of Eclipse RCP),
+ * containing parts covered by the terms of the Eclipse Public License, the licensors of the
+ * ULTIMATE TraceCheckerUtils Library grant you additional permission to convey the resulting work.
+ */
+
+package de.uni_freiburg.informatik.ultimate.lib.tracecheckerutils.assertorders;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import de.uni_freiburg.informatik.ultimate.automata.nestedword.NestedWord;
+import de.uni_freiburg.informatik.ultimate.lib.modelcheckerutils.cfg.structure.IAction;
+import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.HashTreeRelation;
+
+/**
+ * @author Betim Musa (musab@informatik.uni-freiburg.de)
+ * @author Frank Schüssele (schuessf@informatik.uni-freiburg.de)
+ */
+public class AssertOrderUtils {
+	/**
+	 * Returns a set of indices that represents all statements that is present in {@code trace}, but not in
+	 * {@code statementIndices}.
+	 */
+	public static Set<Integer> getTraceDifference(final NestedWord<?> trace, final Set<Integer> statementIndices) {
+		return IntStream.range(0, trace.length()).boxed().filter(x -> !statementIndices.contains(x))
+				.collect(Collectors.toSet());
+	}
+
+	/**
+	 * Partition the statements of the given trace according to their depth.
+	 */
+	public static <LETTER extends IAction, LOC> Map<Integer, Set<Integer>> partitionStatementsAccordingDepth(
+			final NestedWord<LETTER> trace, final HashTreeRelation<LOC, Integer> rwt, final List<LOC> pps) {
+		final Map<Integer, Set<Integer>> depth2Statements = new HashMap<>();
+
+		dfsPartitionStatementsAccordingToDepth(0, trace.length(), 0, rwt, depth2Statements, pps);
+
+		return depth2Statements;
+	}
+
+	/**
+	 * Partition the statement positions between lowerIndex and upperIndex according to their depth. (See documentation
+	 * for the meaning of 'depth'). The result is stored in the map 'depth2Statements'. The partitioning is done
```

Comment:
Where is this documentation regarding the meaning of "depth"?

Reply:
Should be fixed, this part was removed and some documentation added to `partitionStatementsAccordingDepth`.